torch_geometric/data/collate.py

```diff
@@ -96,12 +96,9 @@ def collate(
                 inc_dict[attr] = incs
 
             # Add an additional batch vector for the given attribute:
-            if (attr in follow_batch and isinstance(slices, Tensor)
-                    and slices.dim() == 1):
-                repeats = slices[1:] - slices[:-1]
-                batch = repeat_interleave(repeats.tolist(), device=device)
-                out_store[f'{attr}_batch'] = batch
-                out_store[f'{attr}_ptr'] = cumsum(repeats.to(device))
+            if attr in follow_batch:
+                out_store[f'{attr}_batch'] = _batch(slices, device)
```

Comment:
Can we unify the computation here into a single function? For example, `repeats` is calculated twice otherwise.

Reply:
Done, see `_batch_and_ptr`. I favored readable code instead of unreadable zip magic.